Classes/RequestHandler/AbstractRecordRequestHandler.php

```diff
@@ -262,34 +232,35 @@ protected function formatDataArray(
             $layerCount++;
         }
 
-        if (!$this->isRecordData($data) && (array)$data !== []) {
+        if (!$this->isRecordData($data) && $data !== []) {
```

Comment:
Also here why not use is_array instead of !== []

Reply:
I'm checking if the array is empty, not for the type. The function already receives `$data` as a typed parameter, so it must be an array at this point.